improvements to github exporter docs (#166)
port-labs/port-docs: docs/exporters/bitbucket-exporter/exporting-all-repositories.md

---
sidebar_position: 6
title: Exporting all repositories
---

:::note Prerequisites

- [Please install our Bitbucket app](./installation).

:::

# Exporting all repositories

> By the end of this tutorial, we will have a `microservice` Blueprint that contains the properties `repository URL`, `Workspace` and `Project` which are automatically synced from Bitbucket to Port entity.
> 
> 1. Create a `microservice` Blueprint and `port-app-config.yml` configuration file.
> 
> To export your Bitbucket `repositories` to Port, you can use the following Port Blueprints definitions, and `port-app-config.yml`:

Reviewer: ```suggestion
In this tutorial you will learn how to create a `microservice` Blueprint that contains the properties `repository URL`, `Workspace` and `Project`, which are automatically synced from Bitbucket to Port Entity.

1. Create a `microservice` Blueprint and `port-app-config.yml` configuration file.

To export your Bitbucket `repositories` to Port, you can use the following Port Blueprints definitions, and `port-app-config.yml`:
```

<details>
<summary> Microservice Blueprint </summary>

```json showLineNumbers
{
  "identifier": "microservice",
  "title": "Microservice",
  "icon": "Service",
  "schema": {
    "properties": {
      "url": {
        "title": "URL",
        "format": "url",
        "type": "string"
      },
      "workspace": {
        "type": "string",
        "title": "Project"
      },
      "project": {
        "type": "string",
        "title": "Workspace"
      }
    },
    "required": []
  },
  "mirrorProperties": {},
  "calculationProperties": {},
  "relations": {}
}
```

</details>

> You have to place the `port-app-config.yml` in the repository's root folder.

Reviewer: ```suggestion
Place the `port-app-config.yml` in the repository's root folder.
```
